homeassistant/components/yolink/coordinator.py

```diff
@@ -52,6 +56,14 @@ async def _async_update_data(self) -> dict:
                         - datetime.strptime(device_reporttime, "%Y-%m-%dT%H:%M:%S.%fZ")
                     ).total_seconds()
                     self.dev_online = rpt_time_delta < YOLINK_OFFLINE_TIME
+                try:
+                    dev_external_data_resp = await self.device.get_external_data()
+                    self.device.device_attrs = dev_external_data_resp.data["extData"]
```

Comment:
Why isn't this all done in the `device.fetch_state` call?

We shouldn't update the attributes of an instance directly that is defined by another library. It's the library's responsibility to do that.

Reply:
`device.fetch_state` is use to obtain date inside device, `device.get_external_data` is use to obtain the APP device settings, Should i move `device.get_external_data` call inside the library?

Reply:
Is there a reason why all the state isn't always fetched for the device?